### ISSUE TITLE
RGB controller unable to change color, RGBW controller under to set white LEDs

### ISSUE THREAD

**Issue by GinSoakedBoy** [steve228uk/homebridge-magichome#44]
I have been using this 5 channel controller: [XCSOURCE LD686](https://www.amazon.com/gp/product/B01AA6221S/ref=oh_aui_search_detailpage?ie=UTF8&psc=1) with this RGBWW led strip: [12V 5050 RGBWW](https://www.amazon.com/gp/product/B00K6ZJVP4/ref=oh_aui_search_detailpage?ie=UTF8&psc=1) for the last few months and I've never managed to get the white LEDs to be controlled by this plugin. Also, when I use the home app to set the color at 100% brightness the MagicHome app (which works fine) will show that the brightness is set to 50%.

~~I just purchased this controller: [Opard WiFi RGB LED Controller](https://www.amazon.com/gp/product/B076WXPWL6/ref=oh_aui_detailpage_o01_s00?ie=UTF8&psc=1) using the same led strip and the home app can turn it on and off but not change color or brightness.~~
Edit: Setting the LED strip to RGBW (despite only being RGB) has fixed the issue.

My config is:

   {
       "accessory": "MagicHome",
       "name": "TV Backlight",
       "ip": "192.168.1.111",
       "setup": "RGBWW",
       "purewhite": true
   },
   {
       "accessory": "MagicHome",
       "name": "LED Strip",
       "ip": "192.168.1.112",
       "setup": "RGBW",
       "purewhite": false
   }

I've tried many combinations of configs for the RGBWW controller. Is there anything I'm missing?

Also, I see there are a couple of forks that improve the handling of white LEDs, how do I install the forked plugin to homebridge on my Raspberry Pi?

**Comment by SpeedmaxX** [comment]
Hi, I have a v4 controller. I can switch on and off, but cannot change colors, brightness or warmwhite. In the magichome-app, everything works fine. 

Any ideas?

**Comment by toshibochan** [comment]
@SpeesmaxX Your controller is RGB?
If is RGB set
“setup”: “RGBW”,
“purewhite”: “false”

**Comment by SpeedmaxX** [comment]
Hi, I do have this: 
https://de.aliexpress.com/item/DC-5V-12V-24V-Mini-Bluetooth-WiFi-LED-RGB-RGBW-Controller-iOS-Android-APP-24Key-IR/32815415160.html?spm=a2g0s.9042311.0.0.450b4c4dKjT4ri

as "WIFI RGBW" model

**Comment by toshibochan** [comment]
What is you config.json?

**Comment by SpeedmaxX** [comment]
Like this:

    "accessories": [
        {
            "accessory": "MagicHome",
            "name": "Highboard",
            "ip": "192.168.123.35",
            "setup": "RGBWW",
            "purewhite": true,
            "singleChannel": false
        },
        {
            "accessory": "MagicHome",
            "name": "Vitrine",
            "ip": "192.168.123.69",
            "setup": "RGBWW",
            "purewhite": true,
            "singleChannel": false
        }
     ],

**Comment by toshibochan** [comment]
Try

"accessories": [
    {
        "accessory": "MagicHome",
        "name": "Highboard",
        "ip": "192.168.123.35",
        "setup": "RGBW",
        "purewhie": true
    },
    {
        "accessory": "MagicHome",
        "name": "Vitrine",
        "ip": "192.168.123.69",
        "setup": "RGBW",
        "purewhite": true
    }
 ],

**Comment by gavind87** [comment]
I have a V2 controller with an RGB WW strip, and the same, only able to change the RGB led's but the WW led's stay the same

**Comment by SpeedmaxX** [comment]
Hi, @toshibochan , seems to work now :-) (my first impression) Thank you!

Changed, as suggested:
`"accessories": [
        {
            "accessory": "MagicHome",
            "name": "Highboard",
            "ip": "192.168.123.35",
            "setup": "RGBW",
            "purewhite": true
        },
        {
            "accessory": "MagicHome",
            "name": "Vitrine",
            "ip": "192.168.123.69",
            "setup": "RGBW",
            "purewhite": true
        }
     ],`

**Comment by toshibochan** [comment]
@SpeedmaxX you’re welcome!
Glad is working now!

**Comment by toshibochan** [comment]
@gavind87 what is your config.json?

**Comment by SpeedmaxX** [comment]
@toshibochan
Working: yes, but setting wrong colours. 

Is it relevant to look at the controller if it has rgbw or wgbr or anything like that printed on the surface and to change config.json according to that?

**Comment by cusco** [comment]
Hello all,

I'm having the same issue, where I can turn on/off 
if I bright the brightness to 0, it turns off, at 1% turns on

the controller I have is:
https://www.amazon.co.uk/gp/product/B01J5A7ABW/ref=oh_aui_detailpage_o01_s00?ie=UTF8&psc=1

I tried all forms of combinations of "setup" and "purewhite" but not really sure.
I would like to try and debug, but haven't figured out even where the flux led python is.

also, flux_led installed from pip allows me to change colours...


my config.json has:

`
  "accessories": [
        {
                "accessory"     : "MagicHome"
                ,"name"         : "LED Strip"
                ,"ip"           : "192.168.88.21"
                ,"setup"        : "RBGW"
                ,"purewhite"    : true
        }

`

**Comment by SpeedmaxX** [comment]
I have V4-Controller. Is there anything to take care off?

**Comment by toshibochan** [comment]
@SpeedmaxX when you control in the magic home app work correctly? If not, you connect the cable wrong to the controller. Sometimes the green and red is wrong in the led strip.

**Comment by SpeedmaxX** [comment]
@toshibochan yes, the app works fine. I cables are "switched", is it enough to change setup e.g. from RGBW to GRBW?

**Comment by toshibochan** [comment]
You just swap the cable! Just unscrew the Red and Green and swap and screw again!

**Comment by toshibochan** [comment]
@speedmaxX
Work swapping the cables?

**Comment by SpeedmaxX** [comment]
seems to work - thanx!

**Comment by csigl** [comment]
I use an RGBWW controller and a strip with two rows of LEDs, one row RGB and the other WW. In the Magic Home app I can control them individually, and turn on/off color or white individually or both at the same time. I set my config to RGBWW and purewhite = true, I can adjust color but not able to control the white. Any ideas? If I change to RGBW I can't control either.

**Comment by toshibochan** [comment]
TRY RGBW

**Comment by csigl** [comment]
I did try that before, and if I do so, I cannot even change the color or RGB anymore.

**Comment by toshibochan** [comment]
Make sure your controller IP address don’t change in your router

**Comment by csigl** [comment]
Yeah I did that, that is not the issue. On RGBW the controller does not respond, setting back to RGBWW works again but under no circumstance I get the whites to turn on. 

Maybe I have a non supported controller? I use this: https://www.amazon.de/gp/product/B075X8TJCD with this https://www.amazon.de/FVTLED-Streifen-Lichterkette-Warmweiß-Lichtstreifen/dp/B07FB5Y2TX/

Thanks for your help!

**Comment by toshibochan** [comment]
I never see your controller.
I use RGB but you can buy RGBW to
Nexlux WiFi Wireless LED Smart Controller Alexa Google Home IFTTT Compatible,Working with Android,iOS System,RGBW Strip Lights DC 12V 24V(No Power Adapter Included) https://www.amazon.com/dp/B07D568P54/ref=cm_sw_r_cp_api_U1rRBbX5RYHDT
